### PR TITLE
Added jenkins jobs(python)and java

### PR DESCRIPTION
**Pull request by BasileoLuv**

### CODE REVIEW

**Comment by BasileoLuv** [comment]
Look only python jobs. Java jobs coming soon

File: jnk/jobs/simple-java-service-master-release/config.xml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+<?xml version='1.1' encoding='UTF-8'?>
```
Comment (savermyas) [inline jnk/jobs/simple-java-service-master-release/config.xml]: It's a Java job

File: jnk/jobs/simple-java-service-dev-build/config.xml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+<?xml version='1.1' encoding='UTF-8'?>
```
Comment (savermyas) [inline jnk/jobs/simple-java-service-dev-build/config.xml]: It's a Java job

File: jnk/jobs/python merge/config.xml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  <canRoam>true</canRoam>
+  <disabled>false</disabled>
+  <blockBuildWhenDownstreamBuilding>false</blockBuildWhenDownstreamBuilding>
+  <blockBuildWhenUpstreamBuilding>false</blockBuildWhenUpstreamBuilding>
+  <jdk>JDK 8</jdk>
+  <triggers/>
+  <concurrentBuild>false</concurrentBuild>
+  <builders>
+    <hudson.tasks.Shell>
+      <command>wget http://nexus:8081/repository/pypi/packages/python-calculator/0.4-SNAPSHOT/Python-Calculator-0.4-20190212.075727.tar.gz
```
Comment (savermyas) [inline jnk/jobs/python merge/config.xml]: Snapshot version is still hardcoded.

File: jnk/jobs/simple-python-service-master-release/config.xml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  <blockBuildWhenUpstreamBuilding>false</blockBuildWhenUpstreamBuilding>
+  <jdk>(System)</jdk>
+  <triggers/>
+  <concurrentBuild>false</concurrentBuild>
+  <builders>
+    <hudson.tasks.Shell>
+      <command>git checkout HEAD
+git branch
+git show --summary | awk &apos;{print $2}&apos; | head -1 &gt;newcommit.txt &amp;&amp; python setup.py sdist bdist_wheel
+mv dist/Python-Calculator-0.4-SNAPSHOT.tar.gz dist/Python-Calculator-0.4.tar.gz
```
Comment (savermyas) [inline jnk/jobs/simple-python-service-master-release/config.xml]: what if you have already bumbed your version higher?

**Review by savermyas — CHANGES_REQUESTED** [review]
Address the comments

File: jnk/jobs/python merge/config.xml [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        <hudson.model.StringParameterDefinition>
+          <name>Snapshot_version</name>
+          <description>Python-Calculator-$Snapshot version 
+example: 0.4.1-SNAPSHOT/Python-Calculator-0.4.1-20190215.125629.tar.gz</description>
+          <defaultValue></defaultValue>
+          <trim>false</trim>
+        </hudson.model.StringParameterDefinition>
+        <hudson.model.StringParameterDefinition>
+          <name>Package_version</name>
+          <description>tar xvzf Python-Calculator-$Package_version
```
Comment (savermyas) [inline jnk/jobs/python merge/config.xml]: Why do you need 2 parameters?

File: jnk/jobs/python merge/config.xml [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+  <blockBuildWhenDownstreamBuilding>false</blockBuildWhenDownstreamBuilding>
+  <blockBuildWhenUpstreamBuilding>false</blockBuildWhenUpstreamBuilding>
+  <jdk>JDK 8</jdk>
+  <triggers/>
+  <concurrentBuild>false</concurrentBuild>
+  <builders>
+    <hudson.tasks.Shell>
+      <command>wget http://nexus:8081/repository/pypi/packages/python-calculator/$Snapshot_version
+tar xvzf Python-Calculator-$Package_version
+cd Python-Calculator-0.4.1-SNAPSHOT
```
Comment (savermyas) [inline jnk/jobs/python merge/config.xml]: what if you have already released 0.4.1 and want to proceed with 0.4.2?

File: jnk/jobs/python merge/config.xml [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+git checkout $(cat newcommit.txt)
+git tag 0.4.1
```
Comment (savermyas) [inline jnk/jobs/python merge/config.xml]: the same as above

File: jnk/jobs/simple-python-service-dev-build/config.xml [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+pip install twine
+bash commit.sh
+git show --summary | awk &apos;{print $2}&apos; | head -1 &gt;newcommit.txt &amp;&amp; python setup.py sdist bdist_wheel
+python ./version.py
+python2.7 -m twine upload dist/* --repository-url http://nexus:8081/repository/pypi/ -u admin -p admin123 --skip-existing
+</command>
+    </hudson.tasks.Shell>
+    <org.jenkinsci.plugins.buildnameupdater.BuildNameUpdater plugin="build-name-setter@1.6.9">
+      <buildName>newcommit.txt</buildName>
+      <macroTemplate>#${BUILD_NUMBER}</macroTemplate>
```
Comment (savermyas) [inline jnk/jobs/simple-python-service-dev-build/config.xml]: Commit hash info is not very informative, better to use snapshot version

File: jnk/jobs/simple-python-service-master-release/config.xml [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+      <parameterDefinitions>
+        <hudson.model.StringParameterDefinition>
+          <name>version</name>
+          <description>example: 0.4.1</description>
+          <defaultValue></defaultValue>
+          <trim>false</trim>
+        </hudson.model.StringParameterDefinition>
+        <hudson.model.StringParameterDefinition>
+          <name>jump</name>
+          <description>_SNAPSHOT-py2-none-any
```
Comment (savermyas) [inline jnk/jobs/simple-python-service-master-release/config.xml]: for what?

File: jnk/jobs/simple-python-service-master-release/config.xml [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+  <properties>
+    <com.coravy.hudson.plugins.github.GithubProjectProperty plugin="github@1.29.3">
+      <projectUrl>https://github.com/BasileoLuv/simple-python-service/</projectUrl>
+      <displayName></displayName>
+    </com.coravy.hudson.plugins.github.GithubProjectProperty>
+    <hudson.model.ParametersDefinitionProperty>
+      <parameterDefinitions>
+        <hudson.model.StringParameterDefinition>
+          <name>version</name>
+          <description>example: 0.4.1</description>
```
Comment (savermyas) [inline jnk/jobs/simple-python-service-master-release/config.xml]: How will I release for example 0.4.1-20180215-12345?

File: jnk/jobs/simple-python-service-master-release/config.xml [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+  <blockBuildWhenDownstreamBuilding>false</blockBuildWhenDownstreamBuilding>
+  <blockBuildWhenUpstreamBuilding>false</blockBuildWhenUpstreamBuilding>
+  <jdk>(System)</jdk>
+  <triggers/>
+  <concurrentBuild>false</concurrentBuild>
+  <builders>
+    <hudson.tasks.Shell>
+      <command>git checkout dev
+git checkout master
+git checkout HEAD
```
Comment (savermyas) [inline jnk/jobs/simple-python-service-master-release/config.xml]: 3 checkouts in a row? for what reason?

File: jnk/jobs/simple-python-service-master-release/config.xml [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+git branch
+git show --summary | awk &apos;{print $2}&apos; | head -1 &gt;newcommit.txt &amp;&amp; python setup.py sdist bdist_wheel
+mv dist/Python-Calculator-$version-SNAPSHOT.tar.gz dist/Python-Calculator-$version.tar.gz
+mv dist/Python_Calculator-$version$jump.whl dist/Python-Calculator-$version-py2-none-any.whl
+pip install twine
+pip install wheel
+python2.7 -m twine upload dist/*  --repository-url http://nexus:8081/repository/pypi-releases/ -u admin -p admin123 --skip-existing
+rm -r dist
+git add .
+git commit -m &quot;information about commit&quot;
```
Comment (savermyas) [inline jnk/jobs/simple-python-service-master-release/config.xml]: Informative

File: jnk/jobs/simple-python-service-master-release/config.xml [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+git checkout dev
+bash artifactvers.sh
```
Comment (savermyas) [inline jnk/jobs/simple-python-service-master-release/config.xml]: where is this script?

**Review by savermyas — CHANGES_REQUESTED** [review]
Updates are still needed

File: jnk/jobs/simple-python-service-master-release/config.xml [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+  <blockBuildWhenUpstreamBuilding>false</blockBuildWhenUpstreamBuilding>
+  <jdk>JDK 8</jdk>
+  <triggers/>
+  <concurrentBuild>false</concurrentBuild>
+  <builders>
+    <hudson.tasks.Shell>
+      <command>git checkout master
+git branch
+grep version setup.py &gt;curr-snapshot.txt
+cut -d= -f2 curr-snapshot.txt &gt;2.txt
```
Comment (savermyas) [inline jnk/jobs/simple-python-service-master-release/config.xml]: Use piping for that.

File: jnk/jobs/simple-python-service-master-release/config.xml [hunk 15]
```diff
@@ -0,0 +1,99 @@
...
+cat curr-release.txt
+bash makeurl.sh $versionandtimestamp
+git status
+cd Python-*
+bash releaseornot.sh
+
+python setup.py sdist bdist_wheel
+currsnapshot=$(cat curr-snapshot.txt )
+mv dist/Python-Calculator*.tar.gz dist/Python-Calculator-$currsnapshot.tar.gz
+mv dist/Python_Calculator*.whl dist/Python-Calculator-$currsnapshot-py2-none-any.whl
```
Comment (savermyas) [inline jnk/jobs/simple-python-service-master-release/config.xml]: do you really need both of them? wheel and tar.gz?

File: jnk/jobs/simple-python-service-dev-build/config.xml [hunk 16]
```diff
@@ -0,0 +1,61 @@
...
+      <ignorePostCommitHooks>false</ignorePostCommitHooks>
+    </hudson.triggers.SCMTrigger>
+  </triggers>
+  <concurrentBuild>false</concurrentBuild>
+  <builders>
+    <hudson.tasks.Shell>
+      <command>curl -L https://bootstrap.pypa.io/get-pip.py | python
+pip install --upgrade pip
+pip install twine
+bash script.sh</command>
```
Comment (savermyas) [inline jnk/jobs/simple-python-service-dev-build/config.xml]: what is it?

File: jnk/jobs/simple-python-service-master-release/config.xml [hunk 17]
```diff
@@ -0,0 +1,99 @@
...
+sed &apos;s/^.//&apos; curr-snapshot.txt &gt;2.txt
+mv 2.txt curr-snapshot.txt
+mv curr-snapshot.txt curr-release.txt
+git add curr-release.txt
+git commit -m &quot;Added release version in txt file&quot;
+cat curr-release.txt
+bash makeurl.sh $versionandtimestamp
+git status
+cd Python-*
+bash releaseornot.sh
```
Comment (savermyas) [inline jnk/jobs/simple-python-service-master-release/config.xml]: what is it?

File: jnk/jobs/update version/config.xml [hunk 18]
```diff
@@ -0,0 +1,66 @@
...
+  <canRoam>true</canRoam>
+  <disabled>false</disabled>
+  <blockBuildWhenDownstreamBuilding>false</blockBuildWhenDownstreamBuilding>
+  <blockBuildWhenUpstreamBuilding>false</blockBuildWhenUpstreamBuilding>
+  <jdk>(System)</jdk>
+  <triggers/>
+  <concurrentBuild>false</concurrentBuild>
+  <builders>
+    <hudson.tasks.Shell>
+      <command>bash artifactversion.sh</command>
```
Comment (savermyas) [inline jnk/jobs/update version/config.xml]: what is it?

**Review by savermyas — CHANGES_REQUESTED** [review]
Address the comments

**Pull request closed without merging**